src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -132,7 +132,11 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
 
             details = FlagEvaluationDetails.from(providerEval, key);
             if (details.getErrorCode() != null) {
-                throw ExceptionUtils.instantiateErrorByErrorCode(details.getErrorCode(), details.getErrorMessage());
+                OpenFeatureError error = ExceptionUtils.instantiateErrorByErrorCode(
+                        details.getErrorCode(),
+                        details.getErrorMessage());
+                details.setValue(defaultValue);
```

Comment:
The previous implementation also sets the reason on the FlagEvaluationDetails

Reply:
Yep this is important. The spec dictates it should be set in error conditions. Regardless of what the provider sets, we should set the reason to ERROR.

Reply:
added and also added to the tests for verification